doc/source/train/examples/lightning/lightning_cola_advanced.ipynb

```diff
@@ -0,0 +1,465 @@
+    "Note that we are using Ray Dataset for data ingestion for faster preprocessing here, but you can also continue to use the native `PyTorch DataLoader` or `LightningDataModule`. See {ref}`this example <lightning_mnist_example>`. \n",
+    "\n",
+    ":::\n",
+    "\n",
+    "\n",
+    "Now, call `trainer.fit()` to initiate the training process."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "trainer = LightningTrainer(\n",
+    "    lightning_config=lightning_config,\n",
+    "    run_config=run_config,\n",
+    "    scaling_config=scaling_config,\n",
+    "    datasets={\"train\": ray_datasets[\"train\"], \"val\": ray_datasets[\"validation\"]},\n",
+    "    datasets_iter_config={\"batch_size\": 16},\n",
+    "    preprocessor=preprocessor,\n",
+    ")\n",
+    "result = trainer.fit()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Result(\n",
+       "  metrics={'_report_on': 'validation_end', 'train_loss': 0.05989973247051239, 'matthews_correlation': 0.5175218541439164, 'epoch': 4, 'step': 670, 'should_checkpoint': True, 'done': True, 'trial_id': '5ae4c_00000', 'experiment_tag': '0'},\n",
+       "  path='/home/ray/ray_results/ptl-sent-classification/LightningTrainer_5ae4c_00000_0_2023-04-05_12-45-05',\n",
+       "  checkpoint=LightningCheckpoint(local_path=/home/ray/ray_results/ptl-sent-classification/LightningTrainer_5ae4c_00000_0_2023-04-05_12-45-05/checkpoint_000004)\n",
+       ")"
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "result"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4. Do Batch Inference with a Saved Checkpoint"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now that we have fine-tuned the module, we can load the checkpoint into a BatchPredictor and perform fast inference with multiple GPUs. It will distribute the inference workload across multiple workers when calling `predict()` and run prediction on multiple shards of data in parallel. \n",
+    "\n",
+    "You can find more details in [Using Predictors for Inference](air-predictors)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from ray.train.batch_predictor import BatchPredictor\n",
+    "from ray.train.lightning import LightningCheckpoint, LightningPredictor\n",
+    "\n",
+    "# Use in-memory checkpoint object\n",
+    "checkpoint = result.checkpoint\n",
+    "\n",
+    "# You can also load a checkpoint from disk:\n",
+    "# checkpoint = LightningCheckpoint.from_directory(\"YOUR_CHECKPOINT_DIR\")\n",
```

Comment:
Might be helpful to point out that this the checkpoint directory is what is written in `checkpoint.path`.